src/pages/index.js

```diff
@@ -1,109 +1,212 @@
-previewImagePopup.setEventListeners();
+        renderer: (data) => {
+          const card = createCard(data);
+          sectionEl.addItem(card);
+        },
+      },
+      ".cards__list"
+    );
+    sectionEl.renderItems();
+  })
+  .catch((err) => {
+    console.log(err);
+  });
 
-function handleImageClick(cardData) {
-  previewImagePopup.open(cardData);
+function createCard(data) {
+  const card = new Card({
+    data,
+    cardSelector: "#card-template",
+    currentUserId,
+    handlePreviewImage: () => {
+      imagePopup.open(data);
+    },
+    handleLikeCard: (shouldRemoveLike) => {
+      if (shouldRemoveLike) {
+        api
+          .removeLike(data._id)
+          .then((data) => {
+            card.updateLikes(data.isLiked);
+          })
+          .catch((err) => {
+            console.log(err);
+          });
+      } else {
+        api
+          .addLike(data._id)
+          .then((data) => {
+            card.updateLikes(data.isLiked);
+          })
+          .catch((err) => {
+            console.log(err);
+          });
+      }
+    },
+    handleDeleteCard: (cardId) => {
+      deletePopup.open();
+      deletePopup.setSubmitAction(() => {
+        deletePopup.renderLoading(true);
+        api
+          .deleteCard(cardId)
+          .then(() => {
+            card.deleteClick();
+            deletePopup.close();
+          })
+          .catch((err) => {
+            console.log(err);
+          })
+          .finally(() => {
+            deletePopup.renderLoading(false);
+          });
+      });
+    },
+  });
+  return card.generateCard();
 }
 
-//PopUpWithForm
-const profileEditPopup = new PopupWithForm(
-  "#profile-edit-modal",
-  handleEditProfileSubmit
+const addFormValidator = new FormValidator(addModalForm, formValidationConfig);
+addFormValidator.enableValidation();
+
+const profileFormValidator = new FormValidator(
+  profileEditForm,
+  formValidationConfig
 );
-profileEditPopup.setEventListeners();
-profileEditButton.addEventListener("click", () => {
-  handleProfileEditClick();
-});
+profileFormValidator.enableValidation();
 
-function handleProfileEditClick() {
-  const user = userInfo.getUserInfo();
-  userNameInput.value = user.name;
-  userDescriptionInput.value = user.job;
-  profileEditPopup.open();
-}
+const aviFormValidator = new FormValidator(aviForm, formValidationConfig);
+aviFormValidator.enableValidation();
 
-function handleEditProfileSubmit(inputValues) {
-  userInfo.setUserInfo(inputValues);
-  profileEditPopup.close();
-}
+const deletePopup = new PopupWithConfirmation("#delete-popup");
+deletePopup.setEventListeners();
 
-// Add Card Modal
-const addCardPopup = new PopupWithForm("#new-card-modal", handleAddCardSubmit);
-addCardPopup.setEventListeners();
-addCardButton.addEventListener("click", () => {
-  addCardFormValidator.resetValidation();
-  addCardPopup.open();
+const editFormModal = new PopupWithForm("#modal", (inputValues) => {
+  editFormModal.renderLoading(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```